Adding Jellyfin Migrator Project to the Migration Page (#256)
jellyfin/jellyfin.org: docs/general/administration/migrate.md

# Migrating

> This page explains how to migrate from Emby to Jellyfin as well as how to migrate a Jellyfin installation.

Author: First of all, thanks for reviewing all changes. 

My bad, I thought it was justified because about half of the entire page was about migrating from Emby to Jellyfin. Should I replace "Emby" by "other media servers"?

> This page explains how to migrate from Emby to Jellyfin as well as how to migrate a Jellyfin installation.

Reviewer: I think a sentence like "This page explains how to migrate Jellyfin installations." could apply both to Jellyfin>Jellyfin and Other>Jellyfin migrations.

> This page explains how to migrate from Emby to Jellyfin as well as how to migrate a Jellyfin installation.

Author: In my ears that sounds like "from jellyfin to X" only. I fixed all the other things and included a slightly different wording; maybe that one sounds fine for both of us.

> Jellyfins internal databases are not designed for migrations. Depending on your case there may be ways to work around this, f.ex. by just migrating parts of the data, or because it's possible maintain the same file paths.

Reviewer: The new database schema (jellyfin.db) is designed for migration, we just don't use it for everything yet. So this sentence is untrue. It also doesn't provide relevant information anyway.

Author: I don't want to trash any work here; I know that there are plans to establish a better database structure. However, to the people reading this page, IMO that's not relevant, because their data is still stuck in library.db and many other files with all of their flaws (w.r.t. to migration) - a.k.a. structures that were not designed with migrations in mind.  

So the key message I wanted to get across here is that except for specific cases, the user has no chance of successfully migrating a Jellyfin installation without data loss if he's not using a tool like mine. Or in other words, *the relevant information* is to answer the number one question people looking into this (probably) ask: "Can't I just copy it?"

That being said, maybe this is more accurate in your eyes and likely more clear for the user as well: 

> Jellyfins internal databases can't be copied nor adjusted easily. Depending...

---

<details>
<summary>About jellyfin.db...</summary>

Jellyfin.db requires search-and-replace patching as well because it stores the profile pictures with absolute paths (f.ex. `C:\ProgramData\Jellyfin\Server\config\users\Max\profile.jpg`). In that sense it's not better than the rest of the database. Once again, this might seem harsh/inaccurate from a developer perspective, but from a user perspective its simple: No, you can't copy the file. No, there is not an easy way to adjust it ("not easy" = "a text editor won't do it"). It's not been designed such that either one of these options were possible. 
Also, I don't know if you're familiar with [what I had to do](https://github.com/MMMZZZZ/Jellyfin-Migrator#technical-documentation) to get the migration working, but SQL editor and text editor are not sufficient. 

</details>

Reviewer: Yeah noting that the database files cannot be copied without changes to them is worth mentioning.

> Jellyfins internal databases are not designed for migrations. Depending on your case there may be ways to work around this, f.ex. by just migrating parts of the data, or because it's possible maintain the same file paths.

Reviewer: I would write out "for example" instead of abbreviating it.

Author: Okay, will change it together with the other changes.

If that's not an option for you, there is also a script available to migrate the entire database without data loss, but it's not a turn-key solution.

## Full Database Migration

> [Jellyfin Migrator](https://github.com/MMMZZZZ/Jellyfin-Migrator) is a script that can migrate your entire installation from Windows to Linux or Linux Docker. It can also do other migrations, like f.ex. a reorganization of your media files. All statistics, settings and metadata are preserved. Please check the readme of the project for further details.

Reviewer: ```suggestion
[Jellyfin Migrator](https://github.com/MMMZZZZ/Jellyfin-Migrator) is a third-party script that can migrate your entire installation from Windows to Linux or Linux Docker. It can also do other migrations, like a reorganization of your media files. All statistics, settings and metadata are preserved.
```

- Clearly state that it's third-party to avoid users opening issues in our repositories
- Don't abbreviate "for example" or just remove it
- No need to instruct the user to read the readme. They'll do that anyway if they need to figure out how the tool works :p

Author: > Clearly state that it's third-party to avoid users opening issues in our repositories

Originally, I had that included. But then I saw that the page links to a couple other (apparently) third-party scripts as well without any such note. Therefore I decided to include such a note in the projects readme instead. 
I also opted against adding a disclaimer for all projects linked here because I felt it would disturb the reading more than it helps - especially since this doesn't seem to have been an issue for the other projects?

Other two points are fine for me; going to adjust them.

Reviewer: > but I'd still suggest to add it to all third-party scripts then

I agree, this should happen in other places as well.

[Plex to Jellyfin migration](https://github.com/wilmardo/migrate-plex-to-jellyfin)

> ## Migrating Linux Install to Docker

Author: Not a native speaker, but from what I've been taught and what I can find, verbs in titles should be capitalized, too. One source for this: https://www.grammarly.com/blog/capitalization-in-the-titles/

> ## Migrating Linux Install to Docker

Reviewer: It's not a title for the entire page but for a section and I believe those rules don't apply there. But I'm not a native speaker either :p
